docs/dev/test-architecture.adoc

```diff
@@ -238,10 +244,16 @@ Expect(output).To(ContainSubstring("No file changes detected, skipping build"))
 ----
 +
 
-* If oc, odo or generic library you are looking for is not present in helper package then create a new library function as per the scenario requirement.
+* If oc, odo or generic library you are looking for is not present in helper package then create a new library function as per the scenario requirement. 
++
+
+* Avoid logic implementation within the test files as it unnecessarily overloads the tests. Always create a new function with all the required logics within the helper package. Check for code duplicacy within helper and test files.
```

Comment:
Avoid logic implementation? Bit confused what you mean..

Reply:
Can remove part about `Check for code duplicacy within helper and test files.` that's pretty self-explanatory

Reply:
Maybe we can shorten it to: Avoid unnecessary function implementation within test files. Check to see if there is a helper function already implement

?

Reply:
> Maybe we can shorten it to: Avoid unnecessary function implementation within test files. Check to see if there is a helper function already implement

+1, In addition to this I will also add library function implementation in this point.